SharedSource/Core/RoslynAnnotations/NullableAnnotations.cs

```diff
@@ -68,31 +68,31 @@ namespace System.Diagnostics.CodeAnalysis
 {
   /// <summary>Specifies that null is allowed as an input even if the corresponding type disallows it.</summary>
   [AttributeUsage (AttributeTargets.Field | AttributeTargets.Parameter | AttributeTargets.Property, Inherited = false)]
-  internal sealed class AllowNullAttribute : Attribute
+  public sealed class AllowNullAttribute : Attribute

```

Comment:
that needs to stay internal

Reply:
Did we discuss how the other projects should be getting those attributes?

Reply:
Most likely via a dedicated library, just like JetBrains.Annotations. We cannot make the types in SharedSource public because this will lead to conflicts when embedding the shared source files into multiple projects. It would be possible to use partial classes to make the attributes public only from inside Remotion.dll itself. We might do this for the time being (i.e. until we decide on a separate library) so we don't have to re-embed into every single re-motion project. The problem with embedding in Remotion.dll as public is that this will need to adapt once we target a .net standard version that includes them. 

So, yes, the short term could be to define internal in shared source, expose public in Remotion.dll and rethink/rework in Q3/2020. Either way, it would be a separate PR.

Reply:
> It would be possible to use partial classes to make the attributes public only from inside Remotion.dll itself.

I'm unclear on that part, how would i do that?

Reply:
Similar to `SharedSource/Core/Utilities/AttributeUtility.cs` & `Remotion/Core/Core/Utilities/ArgumentUtility.Partial.cs` I think.